### PR TITLE
Add discounts and totals in the cart

### PR DESCRIPTION
**Pull request by kulczy**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | partially #10261 
| License         | MIT

Zrzut ekranu 2019-05-22 o 13 19 42

### CODE REVIEW

**Comment by CoderMaggie** [comment]
Could we make the "Taxes total" line grey when included in the price?

**Comment by vvasiloi** [comment]
@kulczy & @CoderMaggie, as far as I know, the order can have both included and excluded taxes.
It's probably not something common, but it's technically possible.
Considering that, I think it's better to display the total of both types, then separately the subtotal for included and excluded.

**Comment by kulczy** [comment]
Zrzut ekranu 2019-05-22 o 14 15 31

**Comment by kulczy** [comment]
Zrzut ekranu 2019-05-23 o 11 04 26 kopia

**Comment by vvasiloi** [comment]
Since the Twig extensions in this PR will esentially introduce new public API, I would make these functions more strict in terms of I/O and their names more explicit.

Order tax: https://gist.github.com/vvasiloi/94afdcb3f082900db448c0fcba0fa591
Order items subtotal: https://gist.github.com/vvasiloi/474e5c5aaf201b2d8bf54605d479ba1d

P.S. Didn't test these.

File: src/Sylius/Bundle/ShopBundle/Twig/OrderItemsSubtotalExtension.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+namespace Sylius\Bundle\ShopBundle\Twig;
+
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+
+class OrderItemsSubtotalExtension extends \Twig_Extension
+{
+    /**
+     * {@inheritdoc}
+     */
```
Comment (GSadee) [inline src/Sylius/Bundle/ShopBundle/Twig/OrderItemsSubtotalExtension.php]: Unnecessary docblock

File: src/Sylius/Bundle/ShopBundle/Twig/OrderTaxesTotalExtension.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+use Sylius\Component\Core\Model\AdjustmentInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Order\Model\AdjustmentInterface as BaseAdjustmentInterface;
+
+class OrderTaxesTotalExtension extends \Twig_Extension
+{
+    /**
+     * {@inheritdoc}
+     */
```
Comment (GSadee) [inline src/Sylius/Bundle/ShopBundle/Twig/OrderTaxesTotalExtension.php]: As above

File: src/Sylius/Behat/Page/Shop/Cart/SummaryPageInterface.php [hunk 3]
```diff
@@ -31,7 +31,17 @@ public function getBaseGrandTotal();
     /**
      * @return string
      */
-    public function getTaxTotal();
+    public function getIncludedTaxTotal(): string;
+
+    /**
+     * @return string
+     */
```
Comment (GSadee) [inline src/Sylius/Behat/Page/Shop/Cart/SummaryPageInterface.php]: Unnecessary docblock

**Comment by GSadee** [comment]
Thank you, Szymon! :1st_place_medal:

**Pull request merged**